crates/namada/src/ledger/protocol/mod.rs

```diff
@@ -577,6 +581,11 @@ where
         vp_wasm_cache,
         tx_wasm_cache,
     } = shell_params;
+    if !crate::parameters::is_tx_allowed(wl_storage, tx.code_sechash())
```

Comment:
and here (with a test below)

Reply:
I believe this is the same path of execution we follow for governance proposals, we call `dispatch_tx` which in turn calls `apply_wasm_tx` but than we do this check here so I wonder if we could prevent governance proposals txs from being applied here, because it looks like this function only checks the allowlist, not if we are about to execute something coming from a proposal

Reply:
I changed this in https://github.com/anoma/namada/commit/f288518a09937aadc86c78d8adc475188e07a027